config/initializers/doorkeeper.rb

```diff
@@ -5,6 +5,8 @@
 
   # This block will be called to check whether the resource owner is authenticated or not.
   resource_owner_authenticator do
+    # JP doesn't think this is ever getting called because ActionController::Base has a
+    # before_filter :authenticate_user!
```

Comment:
You are probably right. Well it will get called, but `authenticate_user!` will always just let it pass through. I think you can remove `authenticate_user!` but it might have to return the `current_user` still. (or maybe not , I'm not sure)